ThabangLenonyana/ShopWise
# Backend

Add Grocery List Feature and Enhanced Recommendation System
Changes Overview
Added new models for grocery list functionality
Implemented a hybrid recommendation system combining collaborative and content-based filtering
Added utility functions for product similarity calculations

Added model to store user grocery lists

New Features:
-Recommendation System Implementation
-Added [Recommendation Service]
-Hybrid recommendation approach (60% collaborative, 40% content-based)
Technical Details
Collaborative Filtering
Uses user favorite products to find similar users
Calculates recommendations based on similar users' preferences
Excludes already favorited products
Content-Based Filtering
Uses TF-IDF vectorization on product attributes
Considers product name, description, and category
Calculates cosine similarity for recommendations
Database Changes
Added new tables:
grocery lists
grocery list items
Testing Instructions
Test grocery list creation and management
Verify the recommendation system with:
New users (cold start)
Users with existing favorites
Edge cases (no favorites)
Dependencies
Added scikit-learn for TF-IDF and cosine similarity calculations
Please review and let me know if any changes are needed.
